dev/benchmarks/macrobenchmarks/lib/src/web/bench_paths.dart

```diff
@@ -0,0 +1,6057 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:macrobenchmarks/src/web/recorder.dart';
+
+/// Measure the performance of path construction.
+///
+/// This benchmarks was generated by running flutter gallery and recording
+/// path calls.
+class BenchPathRecording extends RawRecorder {
+  BenchPathRecording() : super(name: benchmarkName);
+
+  static const String benchmarkName = 'bench_path_recording';
+
+  @override
+  Future<void> setUpAll() async {
+  }
+
+  @override
+  void body(Profile profile) {
+    profile.record('recordPathConstruction', () {
+      for (int i = 1; i <= 10; i++) {
+        createPaths();
+      }
+    }, reported: true);
+  }
+}
+
+PathFillType gFillType;
```

Comment:
The top part of this class is hand-maintained. The bottom part is auto-generated. I'd really prefer to separate them into two files, one 100% hand-maintained, the other 100% generated. The generated file should provide instructions for how to regenerate it.

Reply:
You can't regenerate. It is very invasive throughout code base and probably not something we want to maintain.

Reply:
The top part is still something we'll maintain by hand (e.g. we can change the `Recorder` API). Let's split the file into two files.

Reply:
Done.